test/client_reset_test.dart

```diff
@@ -323,50 +333,58 @@ Future<void> main([List<String>? args]) async {
 
       await triggerClientReset(realm, restartSession: false);
       realm.syncSession.resume();
-      await waitFutureWithTimeout(onAfterCompleter.future, timeoutError: "Neither onAfterDiscard, onAfterDiscard nor onManualResetFallback is reported.");
+      await onAfterCompleter.future.wait(defaultWaitTimeout, "Neither onAfterDiscard, onAfterDiscard nor onManualResetFallback is reported.");
     });
   }
 
-  baasTest('Disabled server recovery - onAfterDiscard callback is invoked for RecoverOrDiscardUnsyncedChangesHandler', (appConfig) async {
-    final app = App(appConfig);
-    final user = await getIntegrationUser(app);
+  {
+    final baasAppName = AppNames.flexible;
```

Comment:
We don't have to pass the name of the app to the tests. I think `AppNames.flexible` is taken by default.

Reply:
This change exactly because I don't want to use the default one. I want to make this explicit especially since it is used later when enabling auto recovery